vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -27,7 +27,9 @@ def openstack_handle(options = {})
       raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
       username = options[:user] || self.authentication_userid(options[:auth_type])
-      password = options[:pass] || self.authentication_password(options[:auth_type])
+
+      # the |"" is here to prevent an unhelpful ArgumentError from OpenStackHandle::Handle#connect
+      password = options[:pass] || self.authentication_password(options[:auth_type]) || ""
```

Comment:
Oh, in authentication mixin, the update_authentication method has an options argument that can have `options[:required] = array of field names`, but apparently the only place that one is used from is `app/controllers/ops_controller/settings/rhn.rb`. I think we might want to change that one to use required_credential_fields as well, thoughs?